app/models/namespace.rb

```diff
@@ -225,4 +225,32 @@ def validate_nesting_level
   def restore_routes
     Route.restore(Route.only_deleted.find_by(source_id: id).id, recursive: true)
   end
+
+  def subtract_from_metadata_summary(namespaces, metadata, update_by_one)
+    namespaces.each do |namespace|
+      metadata.each do |metadata_field, value|
+        value = 1 if update_by_one
+        if namespace.metadata_summary[metadata_field] == value
+          namespace.metadata_summary.delete(metadata_field)
+        else
+          namespace.metadata_summary[metadata_field] -= value
```

Comment:
Question: Is it possible for update_by_one to be false and the value not an integer?

Reply:
So this was my design decision and attempt at making the add and subtract metadata summary as generic as possible so that it can be used first by sample metadata updating as shown in this PR, and then also sample deletion/transfer and group/project transfer/deletion in the next PR. 
So anything regarding updating a sample's metadata, where we're dealing with actual sample metadata (ie add `metadatafield1: value1` to a sample) will have `update_by_one` as `true`, because you're right, default `hash value` will possibly not be an integer.
But then when we're dealing with any groups or projects for transfer/deletion, all we care about is their `metadata_summary` (ie: `metadatafield1: 5`), in which case, `update_by_one` is false, and the hash value will be an integer.

I actually initially had separate functions to handle these, but figured they're so similar that I put them together with the `update_by_one` being the varying factor. For readability and simplifying understanding in the future, we could definitely go back to having the functions separated.
Hope that all makes sense

Reply:
Ok. I got confused with update_by_one being false and setting the value. I was also thinking there might be a case were we don't want to add or subtract, but update. Sorry and thanks.